src/masonite/routes/HTTPRoute.py

```diff
@@ -191,6 +197,15 @@ def get_response(self, app=None):
             params = self.extract_parameters(app.make("request").get_path()).values()
             # Resolve Controller Method
             response = app.resolve(getattr(controller, self.controller_method), *params)
+
+            # if request method is HEAD, return a response without content but with
+            # headers that would be returned if request's URL was requested with GET method instead
+            # https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD
+            if app.make("request").get_request_method().upper() == "HEAD":
```

Comment:
Here we should get the response headers and build a new response with the same headers but without content and the 204 status code. Also the response bound to the container is not good anymore we have to replace it.

and here I am not really satisfied on the way to override the current response and build a new one... how could that be written better ?